pkg/machineoutput/types.go

```diff
@@ -54,6 +56,24 @@ func OutputError(machineOutput interface{}) {
 	}
 }
 
+// CatalogListComponents `odo catalog list components -o json` standard machine readable output
+func CatalogListComponents(images []catalog.CatalogImage) {
```

Comment:
This is not good approach :-( 
Catalog should have similar structure like `components`  - see https://github.com/openshift/odo/blob/master/pkg/component/types.go
or `url` https://github.com/openshift/odo/blob/master/pkg/url/types.go

You need to create `types.go`  move `CatalogImage` to it and  create `CatalogImageList` struct and modify `catalog.List`  to return it.